src/main/drivers/vtx_rtc6705.c

```diff
@@ -170,40 +183,77 @@ static void rtc6705Transfer(uint32_t command)
     spiTransferByte(RTC6705_SPI_INSTANCE, (command >> 0) & 0xFF);
 
+    delayMicroseconds(2);
+
     DISABLE_RTC6705;
+
+    delayMicroseconds(2);
 }
 
 /**
  * Set a band and channel
  */
-void rtc6705SetChannel(uint8_t band, uint8_t channel)
+void rtc6705SetChannel(const uint8_t band, const uint8_t channel)
 {
-    band = constrain(band, RTC6705_BAND_MIN, RTC6705_BAND_MAX);
-    channel = constrain(channel, RTC6705_CHANNEL_MIN, RTC6705_CHANNEL_MAX);
+    uint8_t constrainedBand = constrain(band, 0, RTC6705_BAND_COUNT - 1);
+    uint8_t constrainedChannel = constrain(channel, 0, RTC6705_CHANNEL_COUNT - 1);
+
+    spiSetDivisor(RTC6705_SPI_INSTANCE, SPI_CLOCK_SLOW);
 
     rtc6705Transfer(RTC6705_SET_HEAD);
-    rtc6705Transfer(channelArray[band-1][channel-1]);
+    rtc6705Transfer(channelArray[constrainedBand][constrainedChannel]);
 }
 
  /**
  * Set a freq in mhz
  * Formula derived from datasheet
  */
-void rtc6705SetFreq(uint16_t freq)
+void rtc6705SetFreq(const uint16_t freq)
 {
-    freq = constrain(freq, RTC6705_FREQ_MIN, RTC6705_FREQ_MAX);
+    uint16_t constrainedFreq = constrain(freq, RTC6705_FREQ_MIN, RTC6705_FREQ_MAX);
 
     uint32_t val_hex = 0;
 
-    uint32_t val_a = ((((uint64_t)freq*(uint64_t)RTC6705_SET_DIVMULT*(uint64_t)RTC6705_SET_R)/(uint64_t)RTC6705_SET_DIVMULT) % RTC6705_SET_FDIV) / RTC6705_SET_NDIV; //Casts required to make sure correct math (large numbers)
-    uint32_t val_n = (((uint64_t)freq*(uint64_t)RTC6705_SET_DIVMULT*(uint64_t)RTC6705_SET_R)/(uint64_t)RTC6705_SET_DIVMULT) / RTC6705_SET_FDIV; //Casts required to make sure correct math (large numbers)
+    uint32_t val_a = ((((uint64_t)constrainedFreq*(uint64_t)RTC6705_SET_DIVMULT*(uint64_t)RTC6705_SET_R)/(uint64_t)RTC6705_SET_DIVMULT) % RTC6705_SET_FDIV) / RTC6705_SET_NDIV; //Casts required to make sure correct math (large numbers)
+    uint32_t val_n = (((uint64_t)constrainedFreq*(uint64_t)RTC6705_SET_DIVMULT*(uint64_t)RTC6705_SET_R)/(uint64_t)RTC6705_SET_DIVMULT) / RTC6705_SET_FDIV; //Casts required to make sure correct math (large numbers)
 
     val_hex |= RTC6705_SET_WRITE;
     val_hex |= (val_a << 5);
     val_hex |= (val_n << 12);
 
+    spiSetDivisor(RTC6705_SPI_INSTANCE, SPI_CLOCK_SLOW);
+
     rtc6705Transfer(RTC6705_SET_HEAD);
+    delayMicroseconds(10);
+    rtc6705Transfer(val_hex);
+}
+
+void rtc6705SetRFPower(const uint8_t rf_power)
+{
+    uint8_t constrainedRfPower = constrain(rf_power, 0, RTC6705_RF_POWER_COUNT - 1);
+
+    spiSetDivisor(RTC6705_SPI_INSTANCE, SPI_CLOCK_SLOW);
+
+    uint32_t val_hex = 0x10; // write
+    val_hex |= 7; // address
+    uint32_t data = constrainedRfPower == 0 ? (PA_CONTROL_DEFAULT | PD_Q5G_MASK) & (~(PA5G_PW_MASK | PA5G_BS_MASK)) : PA_CONTROL_DEFAULT;
+    val_hex |= data << 5; // 4 address bits and 1 rw bit.
+
     rtc6705Transfer(val_hex);
 }
 
+void rtc6705Disable(void)
+{
+#ifdef RTC6705_POWER_PIN
+    DISABLE_VTX_POWER;
+#endif
+}
+
+void rtc6705Enable(void)
+{
+#ifdef RTC6705_POWER_PIN
+    ENABLE_VTX_POWER;
```

Comment:
I don't like these procedural macros that don't include brackets. So redefined `ENABLE_VTX_POWER` so that `ENABLE_VTX_POWER()` can be used. As well as looking better, it means it can be redefined as a `static inline` function at a later point, if desired.